### ISSUE TITLE
[Expo EAS] Sourcefiles not getting uploaded only for Android

### ISSUE THREAD

**Issue by BrodaNoel** [expo/sentry-expo#292]
### Summary

For some reason, the sourcemap files are getting uploaded for iOS, and not for Android.

The fact that iOS are working well, kind of confirm me that the configuration is correct.

<img width="400" alt="Screen Shot 2022-10-15 at 11 43 03" src="https://user-images.githubusercontent.com/1837969/195992462-1bfee371-c66c-453c-ae3b-c0e33ddf5621.png">

<img width="379" alt="Screen Shot 2022-10-15 at 11 43 09" src="https://user-images.githubusercontent.com/1837969/195992470-6f1bfd8d-7535-4c3b-9a2b-3a21065cf4a4.png">

I have:
```
"expo": "^46.0.0",
"sentry-expo": "~5.0.0",
"@sentry/react-native": "4.2.2",
```

Then in the app.json:
```
"plugins": ["sentry-expo"],
"hooks": {
      "postPublish": [
        {
          "file": "sentry-expo/upload-sourcemaps",
          "config": {
            "organization": "broda-noel",
            "project": "nofilter",
            "authToken": "xxx"
          }
        }
      ]
    },
```

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

46

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 11.6.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.4.0 - ~/.nvm/versions/node/v16.4.0/bin/node
      Yarn: 1.22.19 - ~/.yarn/bin/yarn
      npm: 7.18.1 - ~/.nvm/versions/node/v16.4.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
      Android SDK:
        API Levels: 27, 31
        Build Tools: 23.0.1, 23.0.3, 25.0.3, 26.0.1, 27.0.3, 28.0.3, 29.0.2
        System Images: android-31 | Google Play Intel x86 Atom_64
    IDEs:
      Android Studio: 2021.1 AI-211.7628.21.2111.8309675
      Xcode: 12.5.1/12E507 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^46.0.0 => 46.0.2
      react: 18.0.0 => 18.0.0
      react-native: 0.69.5 => 0.69.5
    npmGlobalPackages:
      eas-cli: 2.4.0
      expo-cli: 6.0.6
    Expo Workflow: managed

### Reproducible demo or steps to reproduce from a blank project

I won't be able to create a Snack for this.

**Comment by BrodaNoel** [comment]
I wonder if the `_` in the package name (only for android) is breaking something.

**Comment by lindesvard** [comment]
Having the exact same issue. Works great on iOS.

In `Run fastlane` I can see `Upload Debug Symbols to Sentry`
But nothing related to uploading debug symbols in `Run gradlew`

**Comment by Norfeldt** [comment]
Is there some way of fixing this until there is a fix from sentry?

tried to run `eas update` to generate the source map and upload it like 


```
sentry-cli releases files "5.0.0 (69)" upload-sourcemaps --rewrite dist/bundles/android-074b9d283bc01e5f7d6fb080b9bf6cce.map dist/bundles/ios-ad37e2eb59be197c7589139d9e01c3e1.map
```

This upload it like

![Screenshot 2022-12-08 at 14 13 45](https://user-images.githubusercontent.com/678806/206455511-7567ad21-3bc5-4ba1-86f4-3730152d61f2.png)

but that does not seem to fix it. I also notice that the iOS postPublish upload looks different:

![Screenshot 2022-12-08 at 14 14 44](https://user-images.githubusercontent.com/678806/206455711-f9b73605-c071-4da4-b89a-0d8b9ad0a578.png)

(it does not have a hash in the file name of the map)

**Comment by fuelkoy** [comment]
Hey @krystofwoldrich, does Sentry show correctly the source map of Sentry.Native.captureException(new Error()); you are throwing? Also I would like to ask that have you run eas update and tried that source maps then work correctly?

**Comment by krystofwoldrich** [comment]
@fuelkoy Thank you for trying it out. From the screenshot, it looks like you uploaded the updated map to the original release. The map name must be `index.android.bundle.map`, but your updated one is `android-6c0d259e381e54ec57d9b286bde31c69.map`, so either you would need to create a new release or overwrite the original `index.android.bundle.map` and `index.android.bundle`.

**Comment by fuelkoy** [comment]
Took my a moment to notice you already answered. But first of all big thank you for helping @krystofwoldrich 

I understood from 'Using Sentry - Expo Documentation' that .map file hash is left there by the script: 
node_modules/@sentry/cli/bin/sentry-cli releases \
    files 'release name' \
    upload-sourcemaps \
    --dist 'Android Update ID' \
    --rewrite \
    dist/bundles/index.android.bundle **dist/bundles/android-'hash'.map.**

Then my problem is that how is the .bundle file uploaded by eas update? I just tested uploading by (and used index.android.bundle.map, not 'hash') and the script uploads only the .map file, not the bundle file. 

Found 1 release file
> Analyzing 1 sources
> Analyzing completed in 0.021s
> Rewriting sources
> Rewriting completed in 0.424s
> Adding source map references
> Bundling files for upload... 
> Bundling completed in 0.682s
> Optimizing completed in 0.025s
> Uploading completed in 7.06s
> Uploaded release files to Sentry
> Processing completed in 0.187s
> File upload complete (processing pending on server)
> Organization: 'organization'
> Project: 'project'
> Release: 'release'
> Dist: 'Android Update ID'

Source Map Upload Report
  Source Maps
    ~/index.android.bundle.map

I opened issue here and posted a StackOverflow question if someone would know why this is happening, but let me so bold to ask you if you would know.

I'm on Windows and using the following script as the one on Using Sentry - documentation didn't work out of the box. Could this script be the culprit as it doesn't find the index.android.bundle file without specifically putting the file extension ( = index.android.bundle.js)

`cross-env ./node_modules/@sentry/cli/bin/sentry-cli releases --org 'organization name' --project 'project name' files 'release name' upload-sourcemaps --dist 'Android Update ID' --rewrite dist/bundles/index.android.bundle dist/bundles/android-'hash'.map`

**Comment by fuelkoy** [comment]
I will double check while writing this. 
And now after checking yes, with `index.android.bundle.js` it works, but then on Sentry the artifact will be `~/index.android.bundle.js`. And on previous test it didn't recognize as it was not `~/index.android.bundle`, but `~/index.android.bundle` (if I understood correctly, I can test this again is needed).
![Sieppaa](https://user-images.githubusercontent.com/65889813/213417420-d9289783-c76b-4721-9d8d-eb0bb88c7854.PNG)

And of course. Here are the links:
Expo: https://docs.expo.dev/guides/using-sentry/#uploading-source-maps-for-updates
Github issue (with more details than in StackOverflow): https://github.com/expo/sentry-expo/issues/313

**Comment by RRaideRR** [comment]
> 

just to get it right @krystofwoldrich : this sample project uploads the Android source maps for you during `eas build` - correct? 

I've run the project with the following dependencies and its not uploading:
```
    "expo": "~48.0.6",
    "expo-status-bar": "~1.4.4",
    "react": "18.2.0",
    "react-native": "0.71.3",
    "sentry-expo": "~6.1.0",
    "expo-application": "~5.1.1",
    "expo-constants": "~14.2.1",
    "expo-device": "~5.2.1",
    "expo-updates": "~0.16.3",
    "@sentry/react-native": "4.13.0"
```
 
 However: iOS upload works. Two minor things that I changed in comparison to your project:
 - I've made use of the SENTRY_AUTH_TOKEN variable
 - I use my own sentry server. 
 
 Thus, my postPublish hook looks like this:
```
     "hooks": {
      "postPublish": [
        {
          "file": "sentry-expo/upload-sourcemaps",
          "config": {
            "organization": "myOrg",
            "project": "expo-test",
            "url": "https://sentry.myServer.com/"
          }
        }
      ]
    },
```

@BrodaNoel @lindesvard - did you manage to get it running?

**Comment by BrodaNoel** [comment]
Nop. 'eas build' still not publishing the sourcemaps

**Comment by lindesvard** [comment]
@RRaideRR sadly not

**Comment by RRaideRR** [comment]
> 

I've tried it and its not working for my setup :-/. Would be nice if you could also test it on your setup.

**Comment by evelant** [comment]
Also seeing this on 6.1.1.

**Comment by ldaudet** [comment]
Same problem here, it's a big regression from the legacy system, and the docs on the expo website are half updated leading to lots of confusion. Seems like there is no way to get eas build to send the sourcemaps to sentry, but even worse is that I can't even find a way to just get the sourcemaps to upload them myself like we have to do for the eas update.

**Comment by CaptainJow** [comment]
I have a similar problem during the build for the apk , I get no errors  but during the build proccess I notice that the size of the uploaded files to EAS is smaller that 200kb and that is not logical .  and the build APK is just the splash screen

**Comment by evelant** [comment]
@brentvatne @kbrandwijk @byCedric @tsapeta Sorry for the notification, but could somebody at Expo bring this to the attention of the appropriate person to get it bumped up in priority? Sentry source maps been broken on Android when building with EAS for nearly 6 months now. That really limits the usefulness of Sentry and gets in the way of the otherwise stellar experience of using EAS. 

If this module is not a priority for active maintenance by Expo perhaps it should be deprecated or spun off into a community repo. As is it's pretty frustrating to have this "official" Expo integration partially broken for such a long time given that the docs say "Sentry treats React Native as a first-class citizen and we have collaborated with Sentry to make sure Expo is, too.".

Thanks!

**Comment by jongbelegen** [comment]
I'm facing the same issue, did anyone use a custom distribution?

At the moment my config looks like this:

```
 hooks: {
      postPublish:
        BUILD_ENV === "dev"
          ? undefined
          : [
              {
                file: "sentry-expo/upload-sourcemaps",
                config: {
                  organization: "<redacted>",
                  project: "<redacted>",
                  authToken: false,
                },
              },
            ],
```

And in my EAS config I have:
```
build.staging.env: {
  "SENTRY_DIST": "embedded"
}
```

In the [sentry docs](https://docs.sentry.io/platforms/react-native/sourcemaps/#using-custom-release-and-distribution) I see the following note:

Source Maps
When you set a custom release and dist in a React Native app, the automatic source maps upload script will no longer work because it does not detect custom values. You will need to [manually upload source maps](https://docs.sentry.io/platforms/react-native/sourcemaps).

**Comment by jongbelegen** [comment]
The reason that this happens for me is that the @sentry/react-native version that is being used is 4.13.0.

This version hooks in by using a task that starts with '[bundle](https://github.com/getsentry/sentry-react-native/blob/514ad4030314183a321a3c868ed782a9c6439fdc/sentry.gradle#L37)',

While in my react native version (0.71.7), I only see the createBundle being ran on build.

It looks like the newer [@sentry/react-native](https://github.com/getsentry/sentry-react-native/blob/5.3.1/sentry.gradle) did change this behavior. I've been trying to upgrade this package. It did upload a source map but other issues arose.

**Comment by giedriusre** [comment]
> @brentvatne @kbrandwijk @byCedric @tsapeta Sorry for the notification, but could somebody at Expo bring this to the attention of the appropriate person to get it bumped up in priority? Sentry source maps have been broken on Android when building with EAS for nearly 6 months now. That really limits the usefulness of Sentry and gets in the way of the otherwise stellar experience of using EAS.
> 
> If this module is not a priority for active maintenance by Expo perhaps it should be deprecated or spun off into a community repo. As is, it's pretty frustrating to have this "official" Expo integration partially broken for such a long time given that the docs say "Sentry treats React Native as a first-class citizen and we have collaborated with Sentry to make sure Expo is, too.".
> 
> Thanks!

Hi. You managed somehow to fix it?

**Comment by giedriusre** [comment]
d

**Comment by krystofwoldrich** [comment]
@jongbelegen Exactly like you say.

We added the support for the new RN bundle tasks in [4.14.0](https://github.com/getsentry/sentry-react-native/releases/tag/4.14.0) version. Maybe try that instead the major update to v5.

Could you elaborate on what other issues arose? Could you share more in the [@sentry/react-native](https://github.com/getsentry/sentry-react-native) repository?

**Comment by giedriusre** [comment]
> 

Sources maps shoud be uploaded automatically while you build app with eas or I need to extra steps for it? Postpublish hooks params I have added in app.json.

**Comment by jongbelegen** [comment]
Hi @krystofwoldrich,

When using this branch: https://github.com/jongbelegen/sentry-expo/tree/fix/update-sentry-android-support

I get the following build error:

```
[RUN_GRADLEW] Source Map Upload Report
[RUN_GRADLEW]   Minified Scripts
[RUN_GRADLEW]     ~/index.android.bundle (sourcemap at index.android.bundle.map)
[RUN_GRADLEW]   Source Maps
[RUN_GRADLEW]     ~/index.android.bundle.map
[RUN_GRADLEW] > Task :app:createBundleReleaseJsAndAssets_SentryCollectModules_com.redacted.app.staging2@2.130.0+embedded_embedded FAILED
[RUN_GRADLEW] node:internal/modules/cjs/loader:1024
[RUN_GRADLEW]   throw err;
[RUN_GRADLEW]   ^
[RUN_GRADLEW] Error: Cannot find module '/private/var/folders/xs/lx_xmc993z183g38nth8zh500000gn/T/eas-build-local-nodejs/03c5a8c7-faae-4d2c-b85f-9134dd055a45/build/redacted-workspace/apps/redacted/node_modules/@sentry/react-native/dist/js/tools/collectModules.js'
[RUN_GRADLEW]     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1021:15)
[RUN_GRADLEW]     at Function.Module._load (node:internal/modules/cjs/loader:866:27)
[RUN_GRADLEW]     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
[RUN_GRADLEW]     at node:internal/main/run_main_module:22:47 {
[RUN_GRADLEW]   code: 'MODULE_NOT_FOUND',
[RUN_GRADLEW]   requireStack: []
[RUN_GRADLEW] }
[RUN_GRADLEW] FAILURE: Build completed with 2 failures.
[RUN_GRADLEW] 1: Task failed with an exception.
```

This probably has to do with using a pnpm monorepo, the package is in the root node_modules and not in de app node_modules.

In my setup I use the custom env variables:
```
SENTRY_RELEASE=events.fastlane.app.staging2@2.130.0
SENTRY_DIST=embedded
```

When I try to use [@sentry/react-native@5.3.1](https://github.com/getsentry/sentry-react-native/tree/5.3.1), it builds correctly and the sourcemaps get uploaded for ios and android. 

![image](https://user-images.githubusercontent.com/11552802/233043566-9b37bce7-2de6-488a-93fc-741efbd6c8eb.png)
(c3b7e303 is the OTA update)

When I trigger an exception it works well on iOS but on android I get the following error on the web ui:

Invalid location in sourcemap (see [Troubleshooting for JavaScript](https://docs.sentry.io/platforms/javascript/sourcemaps/troubleshooting_js/))

When running `sentry-cli sourcemaps explain 12cd99826cb84c5d9ccfb9d7f5289bbd`
I get the following response:

```
✔ Fetched data for event: 12cd99826cb84c5d9ccfb9d7f5289bbd
✔ Event has release name: com.redacted.app.staging2@2.130.0
✔ Event has a valid exception present
✖ Event exception has no stacktrace available
```

I also checked my dist for this triggered event and its "embedded".

What would be handy here? Should I make a small reproducable repo using `eas build --local` with the [@sentry/react-native@5.3.1] dependency?

**Comment by krystofwoldrich** [comment]
@jongbelegen Hi, yes the build error is caused by the mono repo structure, it was fixed in the latter version not to crash the build.
You can adjust the path here: https://github.com/getsentry/sentry-react-native/issues/2740#issuecomment-1396152679

For the Android Stack trace, a minimal reproducible example would help a lot with debugging. Link to a repo or zip would work. :)

**Comment by anirudhsama** [comment]
@krystofwoldrich Where should we configure `collectModulesScript`? In an expo managed app, we cannot modify `gradle.properties` file right?

**Comment by anirudhsama** [comment]
@krystofwoldrich I tried applying the fix by using an expo config, but it still gives the error. Im using this config plugin:

```js
const { withAppBuildGradle } = require("@expo/config-plugins");

// Applying the fix mentioned here https://github.com/getsentry/sentry-react-native/issues/2740

module.exports = (config) => {
	const sentryFix = `project.ext.sentryCli=[collectModulesScript: new File(["node", "--print", "require.resolve('@sentry/react-native/package.json')"].execute().text.trim(), "../dist/js/tools/collectModules.js")]`;

	return withAppBuildGradle(config, (config) => {
		if (config.modResults.contents.includes("project.ext.sentryCli")) {
			return config;
		}

		config.modResults.contents = config.modResults.contents + "\n" + sentryFix;

		return config;
	});
};

```

**Comment by anirudhsama** [comment]
@expo/config-plugins is a way to modify the `app/build.gradle` in expo managed workflow. The above code adds the this snippet in `app/build.gradle` file:

```
project.ext.sentryCli=[
        collectModulesScript: new File(["node", "--print", "require.resolve('@sentry/react-native/package.json')"].execute().text.trim(), "../dist/js/tools/collectModules.js")
]
```

This was required since sentry/react-native is configured in a mono repo and I had to adjust the path of the `colletModules.js` file, as mentioned here: https://github.com/getsentry/sentry-react-native/issues/2740#issuecomment-1396152679.

I will provide a sample project.

Also, this repo (sentry-expo) also uses @expo/config-plugins to provide the integration of sentry/react-native and expo.

**Comment by anirudhsama** [comment]
@krystofwoldrich Sample repo https://github.com/anirudhsama/sentry-expo-demo

**Comment by anirudhsama** [comment]
@maximgeerinck Sure, will try this out! Thanks!

**Comment by mccraveiro** [comment]
@maximgeerinck Thanks for the expo-plugin. It fixed the sourcemap upload. However, It's still not working. Got the following error: 

> There was 1 problem processing this event
> Invalid location in sourcemap (see [Troubleshooting for JavaScript](https://docs.sentry.io/platforms/javascript/sourcemaps/troubleshooting_js/))
> 
> Source `app:///index.android.bundle`

Also, the bundle size is 0 bytes 🤔 

<img width="696" alt="Screenshot 2023-05-11 at 17 38 00" src="https://github.com/expo/sentry-expo/assets/782333/d6011e4d-1198-416d-8436-e3b80430ad78">

**Comment by mccraveiro** [comment]
My issue doesn't seem related. It's also happening on iOS builds. I'll move it to a new issue. 

Thanks again @maximgeerinck ❤️

**Comment by mccraveiro** [comment]
@celoprd not yet 😞 

I've seen [this issue](https://github.com/getsentry/sentry-react-native/issues/3061) that seems related but I couldn't fix on my project.

**Comment by kbrandwijk** [comment]
We have released sentry-expo@6.2.0 which includes a version bump of @sentry/react-native that fixes Android build-time sourcemaps uploads. Feel free to open a new issue if you're still seeing issues with Android build-time sourcemap uploads after upgrading sentry-expo and @sentry/react-native.
